website/integrations: add atlassian (#14209)
goauthentik/authentik: website/integrations/services/atlassian/index.mdx

---
title: Integrate with Atlassian Cloud
sidebar_label: Atlassian Cloud
support_level: community
---

## What is Atlassian Cloud

> Atlassian is a proprietary software company that specializes in collaboration tools designed primarily for software development and project management. Atlassian Cloud is their cloud platform and provides access to their popular apps; Jira, Confluence, Bitbucket, Trello and others.
>
> -- https://www.atlassian.com/

:::important
This guide offers instructions for setting up authentik as a SAML provider specifically for Atlassian Cloud. It is applicable to all Atlassian Cloud applications, including Jira, Confluence, Bitbucket, Trello, and others.

> SAML SSO for Atlassian Cloud apps requires an [Atlassian Guard](https://support.atlassian.com/security-and-access-policies/docs/understand-atlassian-guard/) subscription and a [verified domain](https://support.atlassian.com/user-management/docs/verify-a-domain-to-manage-accounts/). Further information on requirements for SSO can be found in the [Atlassian SSO documentation](https://support.atlassian.com/security-and-access-policies/docs/configure-saml-single-sign-on-with-an-identity-provider/).

Reviewer: Consider moving this paragraph about Atlassian Guard being required down to the Preparation section, because it is a pre-req. It gets a little lost within this big Note.

Atlassian Cloud has two types of users; **internal** and **external**.

Internal users are defined by their email domain which needs to be a [verified domain in Atlassian Cloud](https://support.atlassian.com/user-management/docs/verify-a-domain-to-manage-accounts/). Internal users are able to utilise SSO without Atlassian Cloud credentials.

[External users](https://support.atlassian.com/security-and-access-policies/docs/who-are-external-users/) are required to log into Atlassian Cloud using Atlassian Cloud credentials. They are then prompted for authentik credentials when accessing specific Atlassian Cloud apps like Jira.
:::

## Preparation

The following placeholders are used in this guide:

- `authentik.company` is the FQDN of the authentik installation.

> 

Reviewer: Do we not need to list the FQDNs for the Atlassian apps, or for the Atlassian Cloud?

:::note
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

## authentik configuration

To support the integration of Atlassian Cloud with authentik, you need to create an application/provider pair in authentik.

### Create an application and provider in authentik

1. Log in to authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair. (Alternatively you can first create a provider separately, then create the application and connect it with the provider.)

> - **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings.

Reviewer: Ideally all of these bullet points should be indednted under Step 2.

- **Choose a Provider type**: select **SAML Provider** as the provider type.
- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
    - Temporarily set the **ACS URL** and **Audience** to `https://temp.temp`
    - Set the **Service Provider Binding** to `Post`.
    - Under **Advanced protocol settings**, set an available signing certificate.
- **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

### Download the signing certificate

1. Log into authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Applications** > **Providers** and click on the name of the newly created Atlassian Cloud provider.
3. Under **Download signing certificate** click the **Download** button. The contents of this certificate will be required in the next section.

## Atlassian Cloud configuration

> 1. Log into the [Atlassian administrator portal](https://admin.atlassian.com) as an Atlassian Cloud organization admin.

Reviewer: ```suggestion
1. Log in to the [Atlassian administrator portal](https://admin.atlassian.com) as an Atlassian Cloud organization admin.
```

> 1. Log in to the [Atlassian administrator portal](https://admin.atlassian.com) as an Atlassian Cloud organization admin.

Reviewer: ```suggestion
1. Log in to the [Atlassian administrator portal](https://admin.atlassian.com) as an Atlassian Cloud organization administrator.
```

2. Navigate to **Security** > **Identity providers**.
3. Under **Choose an identity provider** select **Other provider**.
4. Provide a **Directory name** e.g authentik and click **Add**.
5. Click **Set up SAML single sign-on** and then **Next**.
6. Set the following required configurations:
    - **Identity provider Entity ID**: `authentik`
    - **Identity provider SSO URL**: `https://authentik.company/application/saml/<application slug>/sso/binding/redirect/`
    - **Public x509 certificate**: enter the contents of the certificate that was downloaded in the previous section.
7. Click **Add**.
8. You will be shown a **Service provider entity URL** and **Service provider assertion consumer service URL**. Copy both, they will be required in authentik.

> 9. Click **Next**

Reviewer: ```suggestion
9. Click **Next**.
```

10. Under **Link domain** select a verified domain.
11. Click **Stop and save SAML**

## Reconfigure authentik provider

1. Log in to authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Applications** > **Providers** and click the **Edit** icon of the newly created Atlassian Cloud provider.
3. Under **Protocol settgins**, set the following required configurations:
    - **ACS URL**: set the acs url to the copied **Service provider assertion consumer service URL** (e.g. https://auth.atlassian.com/login/callback?connection=saml-example).
    - **Audience**: set the audience to the copied **Service provider entity URL** (e.g. https://auth.atlassian.com/saml/example).
4. Click **Update**

## Enabling SSO in Atlassian Cloud

### Internal users

1. Log into the [Atlassian administrator portal](https://admin.atlassian.com) as an Atlassian Cloud organization admin.
2. Navigate to **Security** > **Authentication policies**.
3. Click **Add policy** at the top right.
4. Select the `authentik` directory and provide a name for the policy.
5. Edit the new policy and check `Enforce single sign-on`.
6. Click **Update**.

### External users

> 1. Log into the [Atlassian administrator portal](https://admin.atlassian.com) as an Atlassian Cloud organization admin.

Reviewer: ```suggestion
1. Log in to the [Atlassian administrator portal](https://admin.atlassian.com) as an Atlassian Cloud organization admin.
```

2. Navigate to **Security** > **External users**.
3. Click on **External user policy**.
4. Under **Authorization method** check **Single sign-on**.
5. Under **Identity provider** select `authentik`.
6. Click **Update**.

## Configuration verification

### Internal users

To verify that authentik is correctly integrated with Atlassian Cloud, first log out of your account. Then, log back in using your credentials for an internal user. You should be redirected to your authentik instance and after successfully logging in, you should be redirected to the selected Atlassian Cloud app.

### External users

To verify that authentik is correctly integrated with Atlassian Cloud, first log out of your account. Then, log back in using your credentials for an external user.

> Once you reach the Atlassian Cloud dashboard, select an app such as Jira. You will be prompted to verify your identity and redirected to your authentik instance. After successfully logging in to authentik you should be logged into the selected Atlassian Cloud app.

Reviewer: ```suggestion
From the Atlassian Cloud dashboard, select an app such as Jira. You will be prompted to verify your identity and redirected to your authentik instance. After successfully logging in to authentik you should be logged into the selected Atlassian Cloud app.
```

---
title: Integrate with Atlassian Cloud
sidebar_label: Atlassian Cloud
support_level: community
---

## What is Atlassian Cloud

> > Atlassian is an Australian-American proprietary software company that specializes in collaboration tools designed primarily for software development and project management. Atlassian Cloud is their cloud platform and provides access to their popular apps; Jira, Confluence, Bitbucket, Trello and others.

Reviewer 2: Do we need to include the fact that it is australian-american?

>
> -- https://www.atlassian.com/

## Important information

> This guide only provides instructions for configuring authentik as a SAML provider for Atlassian Cloud. It applies to all Atlassian Cloud apps including; Jira, Confluence, Bitbucket, Trello, and others.

Reviewer 2: ```suggestion
This guide offers instructions for setting up authentik as a SAML provider specifically for Atlassian Cloud. It is applicable to all Atlassian Cloud applications, including Jira, Confluence, Bitbucket, Trello, and others.
```

> This guide does not provide instructions for SCIM user provisioning, which requires creating a custom Atlassian API integration and is beyond the scope of this document.

Reviewer 2: ```suggestion
This guide does not include instructions for SCIM user provisioning, as it involves creating a custom Atlassian API integration, which is outside the scope of this document.
```

> If this guide is implemented, users are still required to login to Atlassian Cloud using Atlassian credentials. Users are asked for authentik credentials when accessing specific Atlassian apps like Jira.

Reviewer 2: Uhhhhhhh I have no idea
```suggestion
User are still required to login to Atlassian Cloud using Atlassian credentials. Users are asked for authentik credentials when accessing specific Atlassian apps like Jira.
```

Author: I prefer the original, or something similar, because I think users will be surprised at the behavior. I want it to be super clear what the outcome of the guide is. But I'm open to suggestions.

## Preparation

The following placeholders are used in this guide:

- `authentik.company` is the FQDN of the authentik installation.

> SAML SSO for Atlassian Cloud apps requires an [Atlassian Guard](https://support.atlassian.com/security-and-access-policies/docs/understand-atlassian-guard/) subscription and a verified domain. Further information on requirements for SSO can be found in the [Atlassian SSO documentation](https://support.atlassian.com/security-and-access-policies/docs/configure-saml-single-sign-on-with-an-identity-provider/).

Reviewer 2: Can we include this as a warning or also as a notice?

Reviewer 2: Or adding it to Important information (Which could be then transformed to a :::warning with that title thingy)

:::note
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

## authentik configuration

> To support the integration of Atlassian with authentik, you need to create an application/provider pair in authentik.

Reviewer 2: Can we replace instances of Atlassian with Atlassian Cloud to differentiate w/ Atlassian DC as discussed internally?

### Create an application and provider in authentik

1. Log in to authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair. (Alternatively you can first create a provider separately, then create the application and connect it with the provider.)

- **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings.

>     - Note the application slug, it will be required in the `Identity provider SSO URL` in the Atlassian configuration section.

Reviewer 2: ```suggestion
    - Note the application slug, it will be required when filling out the **Identity provider SSO URL** later on.
```

Reviewer 2: Value in the UI so bold. Also, just saying later on seems fine with me as one could presume the next section would be about configuring atlassian

- **Choose a Provider type**: select **SAML Provider** as the provider type.
- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
    - Temporarily Set the **ACS URL** and **Issuer** to `https://temp.temp`
    - Set the **Service Provider Binding** to `Post`.
    - Under **Advanced protocol settings**, select an available signing certificate.
- **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

### Download the signing certificate

> 1. Log in to authentik as an admin, and open the authentik Admin interface.

Reviewer 2: ```suggestion
1. Log into authentik as an admin, and open the authentik Admin interface.
```

I think

2. Navigate to **Applications** > **Providers** and click on the name of the newly created Atlassian provider.
3. Under **Download signing certificate** click the **Download** button. This certificate will be required by Atlassian.

## Atlassian configuration

> 1. Login to admin.atlassian.com as an admin.

Reviewer 2: ```suggestion
1. Log into https://admin.atlassian.com as an admin.
```

> as an admin

are there multiple perms / RBAC? Does the user need to have a specific permission such as "Manage SSO"?

> 1. Login to admin.atlassian.com as an Atlassian organization admin.

Reviewer 2: ```suggestion
1. Log into the [Atlassian administrator portal](https://admin.atlassian.com) as an Atlassian organization admin.
```

> 2. Navigate to **Security** > **Identity providers**.

Reviewer 2: Can we include a note saying that the user has to select Security from the nav bar at the top (I don't remember if there is more than one but mabye worth mentioning?)

Author: After logging into the admin panel you only have 6 options, one of them being security. So I think it's clear enough.

3. Under **Choose an identity provider** select **Other provider**.
4. Provide a **Directory name** e.g authentik and click **Add**.
5. Click **Set up SAML single sign-on** and then **Next**.
6. Set the following required configurations:

>     - **Identity provider Entity ID**: `authentik`

Reviewer 2: Why do we set the issuer/entity id to https://temp.temp on L46 if we set it to authentik here. Couldn't we tell the user to set it to `authentik` before?

    - **Identity provider SSO URL**: `https://authentik.company/application/saml/<application slug>/sso/binding/redirect/`
    - **Public x509 certificate**: enter the contents of the certificate that was downloaded in the previous section.
7. Click **Add**.
8. You will be shown a **Service provider entity URL** and **Service provider assertion consumer service URL**. Copy both, they will be required in authentik.
9. Click **Next**
10. Under **Link domain** select a verified domain.
11. Click **Stop and save SAML**

> ### Add URLs to authentik provider

Reviewer 2: I think we should make this an H2 and say something like "Reconfigure authentik provider"

1. Log in to authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Applications** > **Providers** and click the **Edit** icon of the newly created Atlassian provider.
3. Under **Protocol settgins**, set the following required configurations:

>     - **ACS URL**: set the acs url to the copied **Service provider assertion consumer service URL** (e.g. https://auth.atlassian.com/login/callback?connection=saml-example).

Reviewer 2: I'd remove the examples. In the ATL ui, it is very clear what URL is what

>     - **Audience**: set the audience to the copied **Service provider entitty URL** (e.g. https://auth.atlassian.com/saml/example).

Reviewer 2: > Audience

> Service provider entitty URL

Should be Issuer not audience

Author: The issuer is `authentik`. The audience is that URL from Atlassian. Just double checked my own testing deployment.

4. Click **Update**

> ### Enabling SSO

Reviewer 2: Tried this out for myself and I had to toggle a few more settings to enable SSO for existing users + disable user/pass login. We might want to add a note about that. Under "Authentication policies" > the sso thiingy > members to make existing members log in with SSO and "Authentication policies" > the sso thiingy > first selection menu or something to require sso

1. Login to admin.atlassian.com as an admin.
2. Navigate to **Security** > **External users**.
3. Click on **External user policy**.
4. Under **Authorization method** check **Single sign-on**.
5. Under **Identity provider** select `authentik`.
6. Click **Update**.

## Configuration verification

To confirm that authentik is properly configured with Atlassian Cloud, log out of Atlassian Cloud and log in using your `External User` Atlassian Cloud credentials.

> At the dashboard, select an Atlassian app such as Jira. You will be asked to verify your identity and be redirected to your authentik instance. Enter your authentik credentials and are now logged into the selected Atlassian app.

Reviewer 2: ```suggestion
Once you reach the dashboard, select an Atlassian application such as Jira. You should be prompted to verify your identity and redirected to authentik. After successfully logging in, you should be granted access to the selected Atlassian application.
```

should vs will. What do you think

Reviewer 2: I don't really like my wording either. Open to suggestions
